src/fsharp/FSharp.Build/Microsoft.FSharp.NetSdk.targets

```diff
@@ -68,14 +64,23 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
                                                      '$(_TargetFrameworkVersionWithoutV)' == '4.7') ">true</_FrameworkNeedsValueTupleReference>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="System.ValueTuple" Version="4.*" Condition=" '$(DisableImplicitSystemValueTupleReference)' != 'true' and '$(_FrameworkNeedsValueTupleReference)' == 'true' "></PackageReference>
-    <PackageReference Include="FSharp.Core" Version="$(FSharpCoreImplicitPackageVersion)"  Condition=" '$(DisableImplicitFSharpCoreReference)' != 'true' "></PackageReference>
-  </ItemGroup>
+  <PropertyGroup>
+    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);PackageFSharpDesignTimeTools</TargetsForTfmSpecificContentInPackage>
+  </PropertyGroup>
 
-  <PropertyGroup> 
-    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);PackageFSharpDesignTimeTools</TargetsForTfmSpecificContentInPackage> 
-  </PropertyGroup> 
+  <Target Name="FSharpCorePackageReferences" BeforeTargets="CollectPackageReferences;">
+    <ItemGroup>
+      <FSharpCorePackages Include="@(PackageReference)" Condition=" '%(Identity)' == 'FSharp.Core' " />
+      <PackageReference Remove="FSharp.Core" Condition=" ('$(DisableImplicitFSharpCoreReference)' == 'true' or @(FSharpCorePackages->Count()) &gt; 1) and '%(PackageReference.DefaultInclude)' == 'true' " />
```

Comment:
Can you add parentheses around `condition1 or condition2 and condition3`?  I'd rather be really explicit about what condition will remove the FSharp.Core reference.

Reply:
@brettfo yeppers